imu_sensor_broadcaster/include/imu_sensor_broadcaster/imu_transform.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright 2025 AIT Austrian Institute of Technology GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * Authors: Christoph Froehlich
+ * Adapted from
+ * https://github.com/ros-perception/imu_pipeline/blob/jazzy/imu_transformer/include/imu_transformer/tf2_sensor_msgs.h
+ */
+
+#ifndef IMU_SENSOR_BROADCASTER__IMU_TRANSFORM_HPP_
+#define IMU_SENSOR_BROADCASTER__IMU_TRANSFORM_HPP_
+
+#include <Eigen/Dense>
+#include "sensor_msgs/msg/imu.hpp"
+
+namespace imu_sensor_broadcaster
+{
+/**
+ * @brief Euler to quaternion Z-Y'-X'' convention
+ * https://eigen.tuxfamily.org/dox-devel/group__TutorialGeometry.html
+ */
+Eigen::Quaternion<double> quat_from_euler(double roll, double pitch, double yaw)
+{
+  return Eigen::Quaternion<double>(
+    Eigen::AngleAxisd(yaw, Eigen::Vector3d::UnitZ()) *
+    Eigen::AngleAxisd(pitch, Eigen::Vector3d::UnitY()) *
+    Eigen::AngleAxisd(roll, Eigen::Vector3d::UnitX()));
+}
+
+/**
+ * @brief Transforms a covariance array from one frame to another
+ */
+inline void transformCovariance(
+  const std::array<double, 9> & in, std::array<double, 9> & out, Eigen::Quaternion<double> r)
+{
+  Eigen::Map<const Eigen::Matrix<double, 3, 3, Eigen::RowMajor> > cov_in(in.data());
+  Eigen::Map<Eigen::Matrix<double, 3, 3, Eigen::RowMajor> > cov_out(out.data());
+  cov_out = r * cov_in * r.inverse();
+}
+
+/**
+ * @brief Transforms sensor_msgs::Imu data from one frame to another
+ */
+inline void doTransform(
+  const sensor_msgs::msg::Imu & imu_in, sensor_msgs::msg::Imu & imu_out,
+  const Eigen::Quaternion<double> r)
```

Comment:
```suggestion
inline void doTransform(
  const sensor_msgs::msg::Imu & imu_in, const Eigen::Quaternion<double> &r, sensor_msgs::msg::Imu & imu_out)
```

It's just my habit to have const and inputs at one side of the functions, if we need to leave it how it is, that is totally okay with me